pkg/controllers/user/pipeline/controller/pipelineexecution/deploy.go

```diff
@@ -65,7 +65,15 @@ func (l *Lifecycle) deploy(projectName string) error {
 		return errors.Wrapf(err, "Error create ns")
 	}
 	secret := getPipelineSecret(nsName, token)
-	if _, err := l.secrets.Create(secret); err != nil && !apierrors.IsAlreadyExists(err) {
+	for i := 0; i < 3; i++ {
+		// If project resource quota is enabled, It won't succeed until the resource quota in namespace is synced.
+		// Do retries for this case.
+		if _, err = l.secrets.Create(secret); err == nil || apierrors.IsAlreadyExists(err) {
+			break
+		}
+		time.Sleep(2 * time.Second)
+	}
+	if err != nil && !apierrors.IsAlreadyExists(err) {
 		return errors.Wrapf(err, "Error create secret")
```

Comment:
I think we can improve the error message a bit: `Error creating a pipeline secret`